htlcswitch/switch.go

```diff
@@ -469,6 +478,51 @@ func (s *Switch) SendHTLC(firstHop lnwire.ShortChannelID, attemptID uint64,
 		htlc:           htlc,
 	}
 
+	// Attempt to fetch the target link before creating a circuit so that
+	// we don't leave dangling circuits. The getLocalLink method does not
+	// require the circuit variable to be set on the *htlcPacket.
+	link, linkErr := s.getLocalLink(packet, htlc)
```

Comment:
Seems this actually also fixes a bug (that would occur rarely since the bandwidth hints would report 0 if the link wasn't online?), where we would possibly never clean up a circuit?

Reply:
I didn't look at the routing-level call-graph -- my reasoning was that `SendHTLC` would be called and the link has died before/during with `RemoveLink` being called and so the circuit would still exist.  On start-up, we'd only clean non-`hop.Source` circuits.  I didn't want to make this problem worse by creating a circuit, and then potentially failing if the outbound link was dust-exposed.